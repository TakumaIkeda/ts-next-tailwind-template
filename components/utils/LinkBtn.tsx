import React, { ReactElement } from "react";
import Link from "next/link";
import classNames from "classnames";

type Props = {
  title: string;
  to: string;
  children?: ReactElement;
};

const LinkBtn: React.FC<Props> = ({ title, to }: Props) => {
  const btn_class = classNames(
    "transition",
    "rounded",
    "text-white",
    "py-2",
    "px-4",
    "m-5",
    "bg-blue-500",
    "hover:bg-blue-600",
    "focus:bg-blue-700"
  );
  return (
    <Link href={to}>
      <a className={btn_class}>{title}</a>
    </Link>
  );
};

export default LinkBtn;
