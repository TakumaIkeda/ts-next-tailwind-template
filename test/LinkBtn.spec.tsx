import React from "react";
import renderer from "react-test-renderer";
import LinkBtn from "../components/utils/LinkBtn";

test("AppTitle", () => {
  const component = renderer.create(<LinkBtn to="/" title="Home" />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
