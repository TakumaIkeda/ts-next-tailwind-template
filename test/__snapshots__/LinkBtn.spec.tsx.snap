// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTitle 1`] = `
<a
  className="transition rounded text-white py-2 px-4 m-5 bg-blue-500 hover:bg-blue-600 focus:bg-blue-700"
  href="/"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  Home
</a>
`;
